src/Workspaces/CSharp/Portable/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -574,6 +567,39 @@ private static bool AtEndOfIncompleteStringOrCharLiteral(SyntaxToken token, int
             }
         }
 
+        private static bool IsSelected(TextSpan textSpan, MemberDeclarationSyntax member, bool allowPartialSelection)
+        {
+            // first, check if entire member is selected
+            if (textSpan.Contains(member.Span))
+            {
+                return true;
```

Comment:
We don't validate if this is a field/property here? That would violate the contract of the caller that only selected fields/properties are returned.

Reply:
Good point. Fixed.